### PR TITLE
Add check for Puppet errors

### PR DESCRIPTION
**Pull request by grem11n**
## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [x] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass 

#### New Plugins

- [ ] Tests

- [x] Add the plugin to the README

- [x] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

#### Purpose
Several use cases:
* You want to check only if Puppet has errors during it's run regardless time of the execution
* You want to split alerting for Puppet issues e.g. send Sensu alert to one subscription if Puppet has errors, and to another subscription if it ran a while ago, but without errors.

#### Known Compatibility Issues
Tested on Ubuntu 16.04. Should work on other Linux distributions as well

### CODE REVIEW

File: bin/check-puppet-errors.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        critical "#{config[:summary_file]} is missing information about the events"
+      end
+    rescue
+      unknown "Could not process #{config[:summary_file]}"
+    end
+
+    @message = 'Puppet run'
+
+    begin
+      disabled_message = JSON.parse(File.read(config[:agent_disabled_file]))['disabled_message']
```
Comment (majormoses) [inline bin/check-puppet-errors.rb]: might be best to check the file existence before parsing it.

File: bin/check-puppet-errors.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      @message += " (disabled reason: #{disabled_message})"
+    rescue # rubocop:disable HandleExceptions
```
Comment (majormoses) [inline bin/check-puppet-errors.rb]: It is best to send the rescued output to a variable to at least print it out: `rescue => e` .

File: bin/check-puppet-errors.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      # fail silently
```
Comment (majormoses) [inline bin/check-puppet-errors.rb]: let's print the failure.

**Review by majormoses — CHANGES_REQUESTED** [review]
Overall looks good, some minor enhancements.

File: bin/check-puppet-errors.rb [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'yaml'
+require 'json'
+require 'time'
+
+class PuppetErros < Sensu::Plugin::Check::CLI
```
Comment (eheydrick) [inline bin/check-puppet-errors.rb]: typo. Erros -> Errors

**Review by eheydrick — CHANGES_REQUESTED** [review]

File: bin/check-puppet-errors.rb (hunk 1 above)
Comment (grem11n) [reply bin/check-puppet-errors.rb]: File existence check is on lines 49-51:
```
    unless File.exist?(config[:summary_file])
       unknown "File #{config[:summary_file]} not found"
      end
```
I can move it to the `begin-end` block, though

**Comment by grem11n** [comment]
Made a few changes. However, I need some more testing for exception handling part for "disabled.lock":

```
    begin
      disabled_message = JSON.parse(File.read(config[:agent_disabled_file]))['disabled_message']
      @message += " (disabled reason: #{disabled_message})"
    rescue => e
      unknown "Could not get disabled message. Reason: #{e.message}"
    end```

It seems like a silent exception was a "workaround" to continue execution if the lock file doesn't exist. Let  me run a few more tests and update this PR

Regards!

File: bin/check-puppet-errors.rb [hunk 5]
```diff
@@ -32,7 +32,7 @@
 require 'json'
 require 'time'
 
-class PuppetErros < Sensu::Plugin::Check::CLI
+class PuppetErrors < Sensu::Plugin::Check::CLI
```
Comment (majormoses) [inline bin/check-puppet-errors.rb]: :100:

File: bin/check-puppet-errors.rb (hunk 1 above)
Comment (majormoses) [reply bin/check-puppet-errors.rb]: nope that is fine I must have missed that.

**Comment by grem11n** [comment]
Should be fine now

**Review by majormoses — APPROVED** [review]

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-puppet/versions/1.3.0

**Pull request merged**